crates/key-server/src/server.rs

```diff
@@ -472,6 +483,23 @@ async fn handle_fetch_key(
     app_state.metrics.requests.inc();
     app_state.check_full_node_is_fresh(ALLOWED_STALENESS)?;
 
+    // Check the SDK version. Fail if the version is not provided or invalid.
+    version
+        .ok_or(InvalidSDKVersion)
+        .and_then(|v| v.to_str().map_err(|_| InvalidSDKVersion))
+        .and_then(|v| app_state.validate_sdk_version(v))
+        .tap_err(|e| {
+            warn!("Error({}): {:?}", e.as_str(), version);
+            app_state.metrics.observe_error(e.as_str());
+        })?;
+
+    // Add the server's version to the response header.
+    let mut headers = HeaderMap::new();
```

Comment:
let's return the version in response headers and verify the min sdk version in all endpoints

Reply:
Got it, but isn't that exactly what it's doing now?